### PR TITLE
Use tox on Jenkins to run against multiple Python versions

### PR DESCRIPTION
**Pull request by mattclarke**
On Jenkins, it will use tox to run the unit tests against multiple versions of Python.
The easiest way to get this to work was install Conda as part of the process.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -4,7 +4,7 @@ import ecdcpipeline.PipelineBuilder
 
 project = "python-streaming-data-types"
 
-python = "python3.6"
+python_version = "3.8"
```
Comment (matthew-d-jones) [inline Jenkinsfile]: I don't think this is used, and given that tox tests against several versions is probably just misleading?

File: Jenkinsfile [hunk 2]
```diff
@@ -37,20 +37,33 @@ builders = pipeline_builder.createBuilders { container ->
     container.copyTo(pipeline_builder.project, pipeline_builder.project)
   }  // stage
 
+  pipeline_builder.stage("${container.key}: Conda") {
+    container.sh """
+      curl https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh --output miniconda.sh
```
Comment (matthew-d-jones) [inline Jenkinsfile]: Should we have conda in the docker image instead of downloading every time?

Comment (mattclarke) [reply Jenkinsfile]: I was thinking the same thing as I will need this for just-bin-it and NICOS at some point.
@amues how does this sound to you?

Comment (amues) [reply Jenkinsfile]: This sounds like a good idea. I'll add it to the image.

Comment (amues) [reply Jenkinsfile]: Miniconda is now available at _/opt/miniconda_ in the version 5.1.0 of the image. The `update` step is run in the image build, but `conda init bash` is not.

Comment (mattclarke) [reply Jenkinsfile]: Thanks, I'll update the jenkinsfile to try it out

Comment (amues) [reply Jenkinsfile]: I had forgotten to update the Jenkins shared library to use the new version of the container. I have just done that and will trigger a new build.

**Comment by amues** [comment]
@mattclarke please use the centos7-gcc8 build node image, as it is the one with Miniconda installed.

**Comment by amues** [comment]
Ah, sorry, I had pushed to the wrong branch of the shared library. This should be fixed now.

**Comment by mattclarke** [comment]
Thanks @amues

**Pull request merged**